src/quickgui/plugin/editor/qgsquickexternalresource.qml

```diff
@@ -97,7 +121,7 @@ Item {
 
       MouseArea {
         anchors.fill: parent
-        onClicked: externalResourceHandler.previewImage(homePath + "/" + image.currentValue)
+        onClicked: externalResourceHandler.previewImage(prefixToRelativePath + "/" + image.currentValue)
```

Comment:
handling of absolute paths is missing (should be just image.currentValue). I would suggest to wrap it into a function that would return either "prefix/image.currentValue" or "image.currentValue" based on the absolute/relative path settings